ui/decredmaterial/scrollbar.go

```diff
@@ -2,163 +2,163 @@
-	lengthOffset := int((windowLength - scrollbarLength) * (contentOffset / (contentLength - windowLength)))
-
-	minY := float32(lengthOffset)
-	maxY := minY + scrollbarLength
-	if maxY > float32(maxSize.Y) {
-		maxY = float32(maxSize.Y)
-		minY = maxY - scrollbarLength
-	}
-
-	thumb := f32.Rectangle{
-		Min: f32.Point{
-			Y: minY,
-		},
-		Max: f32.Point{
-			Y: maxY,
-			X: float32(maxSize.X),
-		},
-	}
-	rr := 0.5 * float32(maxSize.X)
-	clip.RRect{
-		Rect: thumb,
-		NE:   rr, NW: rr, SE: rr, SW: rr,
-	}.Add(gtx.Ops)
-	fill(gtx, color)
-
-	return layout.Dimensions{Size: maxSize}
-}
-
-func (s *Scrollbar) Position() float32 {
-	return s.position
-}
-
-type ScrollContainer struct {
-	container                       *layout.List
-	scrollbar                       Scrollbar
-	totalContentHeight              float32
-	hasCalculatedTotalContentHeight bool
-	scrollbarThicknessPercentage    float32
-}
-
-func (t *Theme) ScrollContainer() *ScrollContainer {
-	return &ScrollContainer{
-		container:                       &layout.List{Axis: layout.Vertical},
-		scrollbar:                       t.Scrollbar(new(Float), 0, 100),
-		totalContentHeight:              0,
-		hasCalculatedTotalContentHeight: false,
-		scrollbarThicknessPercentage:    1.3,
-	}
-}
-
-func (s *ScrollContainer) calculateTotalContentHeight(gtx layout.Context, widgets []layout.Widget) {
-	for i := range widgets {
-		index := i
-		(&layout.List{Axis: layout.Vertical}).Layout(gtx, 1, func(gtx C, i int) D {
-			dim := layout.UniformInset(unit.Dp(0)).Layout(gtx, widgets[index])
-			s.totalContentHeight += float32(dim.Size.Y)
-			return layout.Dimensions{}
-		})
-	}
-	s.hasCalculatedTotalContentHeight = true
-}
-
-func (s *ScrollContainer) Layout(gtx layout.Context, widgets []layout.Widget) layout.Dimensions {
-	maxSize := gtx.Constraints.Max
-
-	if s.scrollbar.scrolled() {
-		s.container.Position.First = int(float32(len(widgets)) * (s.scrollbar.Position() / float32(maxSize.Y)))
-	}
-
-	totalVisibleHeight := float32(maxSize.Y)
-	scrollbarThickness := (s.scrollbarThicknessPercentage / 100) * float32(1)
-
-	return layout.Flex{}.Layout(gtx,
-		layout.Flexed(1-scrollbarThickness, func(gtx C) D {
-			if !s.hasCalculatedTotalContentHeight {
-				s.calculateTotalContentHeight(gtx, widgets)
-				return layout.Dimensions{}
-			}
-
-			return layout.Inset{
-				Right: unit.Dp(15),
-			}.Layout(gtx, func(gtx C) D {
-				return s.container.Layout(gtx, len(widgets), func(gtx C, i int) D {
-					return layout.UniformInset(unit.Dp(0)).Layout(gtx, widgets[i])
-				})
-			})
-		}),
-		layout.Flexed(scrollbarThickness, func(gtx C) D {
-			// don't display scrollbar if total content height is less than or equal to container height
-			if s.totalContentHeight <= totalVisibleHeight {
-				return layout.Dimensions{}
-			}
-
-			initialPosition := float32(s.container.Position.First) / float32(len(widgets))
-			return s.scrollbar.Layout(gtx, s.totalContentHeight, initialPosition)
-		}),
-	)
-}
+// import (
```

Comment:
why is this commented?

Reply:
The reason for the Library update was to remove this custom widget.

Reply:
This has been implemented on the gioui library